src/cli/list_themes.zig

```diff
@@ -192,6 +192,31 @@ pub fn run(gpa_alloc: std.mem.Allocator) !u8 {
     return 0;
 }
 
+fn resolveAutoThemePath(alloc: std.mem.Allocator) ![]u8 {
+    const main_cfg_path = try Config.preferredDefaultFilePath(alloc);
+    defer alloc.free(main_cfg_path);
+
+    const base_dir = std.fs.path.dirname(main_cfg_path) orelse return error.BadPathName;
+    return try std.fs.path.join(alloc, &.{ base_dir, "auto", "theme.ghostty" });
+}
+
+fn writeAutoThemeFile(alloc: std.mem.Allocator, theme_name: []const u8) !void {
+    const auto_path = try resolveAutoThemePath(alloc);
+    defer alloc.free(auto_path);
+
+    if (std.fs.path.dirname(auto_path)) |dir| {
+        try std.fs.cwd().makePath(dir);
+    }
+
+    var f = try std.fs.createFileAbsolute(auto_path, .{ .truncate = true });
```

Comment:
Rather than directly writing the file out, I'd use the [atomicFile](https://ziglang.org/documentation/0.15.2/std/#std.fs.Dir.atomicFile) interface to ensure that the file is always kept consistent.